### PR TITLE
Clarify that MetricReader has one-to-one mapping to MeterProvider

### LINKED ISSUE #2072: Clarification on MetricReader registered on MeterProvider
**What are you trying to achieve?**

Clarify the object ownerships between metric reader and meter provider.

In the example (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view), it is said the metric reader is constructed in place and then it is registered to the meter provider:

```python
meter_provider
    .add_view("*") # a wildcard view that matches everything
    .add_metric_reader(PeriodicExportingMetricReader(ConsoleExporter()))
```

Also, it is stated that metric reader has a `Collect` operation that triggers asynchronous instrument callbacks:

> Metric Reader Operations:
> Collect: Collects the metrics from the SDK. If there are asynchronous Instruments involved, their callback functions will be triggered.
> https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricreader-operations

As far as I could tell, all the instruments are owned by the meter provider, and thus the `collect` should consult the meter provider to perform async instrument callbacks. 

In this sense, it seems that the reader can be registered to multiple meter provider and the `metric_reader.collect` should perform collection on all registered meter provider:

```python
metric_reader = PeriodicExportingMetricReader(ConsoleExporter())
meter_provider
    .add_view("*")
    .add_metric_reader(metric_reader)

meter_provider_b
    .add_view("*")
    .add_metric_reader(metric_reader)
```

The problem here is that the SDK API interface suggested that we are registering the reader to the meter provider, however, the `metric_reader.collect` reverse the control ownership and the meter provider shall be registered on the metric reader and thus it is possible to call `metric_reader.collect` to perform a collection on all registered async instrument callbacks.

This sounds like a doubly linked ownership:
```
        meter_provider 
  shutdown/    |  ↑
  force_flush  |  | collect
               ↓  |
        metric reader
```

**What did you expect to see?**

The ownership should not be bidirectional.

### PR DESCRIPTION
**Pull request by reyang**
Fixes #2072.

### CODE REVIEW

**Review by legendecas — APPROVED** [review]
👍

**Review by jmacd — APPROVED** [review]

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -652,6 +652,9 @@ that SDK will update the time range - e.g. from (T<sub>n</sub>, T<sub>n+1</sub>]
 to (T<sub>n+1</sub>, T<sub>n+2</sub>] - **ONLY** for this particular
 `MetricReader` instance.
 
+The SDK MUST NOT allow a `MetricReader` instance to be registered on more than
+one `MeterProvider` instances.
```
Comment (pirgeo) [inline specification/metrics/sdk.md:656]:
```suggestion
one `MeterProvider` instance.
```

**Review by pirgeo — APPROVED** [review]

File: specification/metrics/sdk.md [hunk 2]
```diff
@@ -652,6 +652,9 @@ that SDK will update the time range - e.g. from (T<sub>n</sub>, T<sub>n+1</sub>]
 to (T<sub>n+1</sub>, T<sub>n+2</sub>] - **ONLY** for this particular
 `MetricReader` instance.
 
+The SDK MUST NOT allow a `MetricReader` instance to be registered on more than
+one `MeterProvider` instance.
```
Comment (carlosalberto) [inline specification/metrics/sdk.md:656]: Does this mean the SDK has to keep track of the `MetricProvider`/`MetricReader` instances it has created?

(I'd have expected that the user MUST NOT re-use a `MetricReader` instead)

Comment (legendecas) [reply specification/metrics/sdk.md:656]: AFAICT, the SDK already needs to keep track of metric readers to track their last collection state for aggregation, as the spec is defined that each metric reader's `collect` has no side effects on others.

Comment (reyang) [reply specification/metrics/sdk.md:656]: > Does this mean the SDK has to keep track of the `MetricProvider`/`MetricReader` instances it has created?

Not necessarily. For example, a certain language implementation could choose to implement this by only allowing `MetricReader` to be provided via a factory method, in this way the SDK could focus on tracking the `MeterProvider`s, and each `MeterProvider` can track their `MetricReader`s.

But from concept level, YES - each MetricReader is essentially having its own state.

**Review by carlosalberto — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request merged**